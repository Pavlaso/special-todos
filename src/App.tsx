import { useState} from 'react';
import {v1} from "uuid";

import { AddItemForm } from "./components/AddItemForm";
import { Buttons } from './components/Buttons';
import { Tasks } from './components/Tasks';

//Тут нет тестирования, так как я не умею писать тесты :), но я готов учится

export const App = () => {

    const [tasks, setTasks] = useState<taskType[]>([])
    const [number, setNumber] = useState(0)
    const [filter, setFilter] = useState('all')


    const deleteTaskHandler= () =>  setTasks([...tasks.filter((t: taskType) => !t.isDone)])

    const changeFilter =(filter: string) => setFilter(filter)

    const addTask=(title: string) => {
        if(title) {
            setTasks([...tasks, { 
                id: v1(), title, isDone: false 
            }])
        }
    }

    const changeStatusTask =(id: string, isDone:boolean) => {
        let arr =[...tasks.map(t => t.id === id ?{...t,isDone: isDone} : t)]
        let activeTasks = [...arr.filter(t => !t.isDone)]
        setTasks(arr)
        setNumber(activeTasks.length)  
    }   

   
    return (
        <div className="App">
            
            <h1 className="logo">todos</h1> 

            <AddItemForm addTask={addTask}/>

            <Tasks tasks={tasks} changeStatusTask={changeStatusTask} filter={filter} />
            
            <Buttons deleteTaskHandler={deleteTaskHandler} changeFilter={changeFilter} filter={filter}  number={number}/>
        </div>
  );
}

export type taskType = {
    id: string
    title: string,
    isDone: boolean
}
