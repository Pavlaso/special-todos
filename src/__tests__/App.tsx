import { render, screen } from '@testing-library/react';
import { App } from '../App';


describe('App', () => {
  it('render logo text', () => {
    render(<App />)
    expect(screen.getByText(/todos/i)).toBeInTheDocument();
  })
  it('App snapshot', () => {
    const app = render(<App />)
    expect(app).toMatchSnapshot()
  })
  
})
