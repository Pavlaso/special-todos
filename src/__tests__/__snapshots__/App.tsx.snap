// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h1
          class="logo"
        >
          todos
        </h1>
        <div
          class="input_task"
        >
          <input
            placeholder="What needs to be done?"
            value=""
          />
          <button>
             Add 
          </button>
        </div>
        <div
          class="list"
        />
        <div
          class="menu"
        >
          <div
            class="sum"
          >
            0
             items left
          </div>
          <div>
            <span
              class="active"
            >
              All
            </span>
            <span
              class=""
            >
              Active
            </span>
            <span
              class=""
            >
              Completed
            </span>
          </div>
          <div>
            <span>
              clear completed
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h1
        class="logo"
      >
        todos
      </h1>
      <div
        class="input_task"
      >
        <input
          placeholder="What needs to be done?"
          value=""
        />
        <button>
           Add 
        </button>
      </div>
      <div
        class="list"
      />
      <div
        class="menu"
      >
        <div
          class="sum"
        >
          0
           items left
        </div>
        <div>
          <span
            class="active"
          >
            All
          </span>
          <span
            class=""
          >
            Active
          </span>
          <span
            class=""
          >
            Completed
          </span>
        </div>
        <div>
          <span>
            clear completed
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
